src/SmartComponents/DriftPage/FilterDropDown/FilterDropDown.js

```diff
@@ -17,77 +16,57 @@ class FilterDropDown extends Component {
+                data-ouia-component-id={ `${ type }-filter-option-${ filter.display }` } >
                 <Checkbox
-                    id={ stateFilter.display }
+                    id={ filter.display }
                     data-ouia-component-type='PF4/Checkbox'
-                    data-ouia-component-id={ 'state-filter-option-checkbox-' + stateFilter.display }
-                    label={ stateFilter.display }
-                    isChecked={ stateFilter.selected }
-                    onChange={ () => this.addStateFilter(stateFilter) }
+                    data-ouia-component-id={ `${ type }-filter-option-checkbox-${ filter.display }` }
+                    label={ filter.display }
+                    isChecked={ filter.selected }
+                    onChange={ () => this.addFilter(filter) }
                 />
             </DropdownItem>;
         return dropdownItem;
     }
 
-    createDropdownArray(stateFilters) {
+    createDropdownArray(filters, type) {
         let dropdownItems = [];
 
-        stateFilters.forEach(function(stateFilter) {
-            let dropdownItem = this.createDropdownItem(stateFilter);
+        filters.forEach(function(filter) {
+            let dropdownItem = this.createDropdownItem(filter, type);
             dropdownItems.push(dropdownItem);
         }.bind(this));
 
         return dropdownItems;
     }
 
-    createSelectedViewsString(stateFilters) {
-        let selectedViewsArray = [];
-        let selectedViews = '';
-
-        for (let i = 0; i < stateFilters.length; i++) {
-            if (stateFilters[i].selected) {
-                selectedViewsArray.push(stateFilters[i].display);
-            }
-        }
-
-        for (let i = 0; i < selectedViewsArray.length; i++) {
-            selectedViews += selectedViewsArray[i];
-
-            if ((i + 1) < selectedViewsArray.length) {
-                selectedViews += ', ';
-            }
-        }
-
-        return selectedViews;
-    }
-
     render() {
-        const { stateFilters } = this.props;
+        const { filters, type } = this.props;
         let dropdownItems = [];
-        let selectedViews = '';
 
-        dropdownItems = this.createDropdownArray(stateFilters);
-        selectedViews = this.createSelectedViewsString(stateFilters);
+        dropdownItems = this.createDropdownArray(filters, type);
 
         return (
             <React.Fragment>
                 <Dropdown
-                    ouiaId='state-filter-dropdown'
+                    ouiaId={ `${ (type.split(' ').join('-')).toLowerCase() }-filter-dropdown` }
                     toggle={ <DropdownToggle
                         onToggle={ this.onToggle }
-                        ouiaId='state-filter-dropdown-toggle' >
-                                    View: { selectedViews }
+                        ouiaId={ `${ (type.split(' ').join('-')).toLowerCase() }-filter-dropdown-toggle` } >
```

Comment:
I have some code repetition on line 62 and 65. It can be unified with a single const string. But this is really minor you can  ignore this if you do not think important